objects/c4e8cd/script.lua

```diff
@@ -1 +1,977 @@
+    local passDirs = {"left", "right"}
+    local passDir = passDirs[round]
+    return destinyPlayers[zoneGUID][passDir]
+end
+
+function getNumPlayers()
+    local num = 0
+    for _,data in pairs(destinyPlayers) do
+        if data then
+            num = num + 1
+        end
+    end
+    return num
+end
+
+function getButtonScale(obj)
+    local scale = obj.getScale()
+    return Vector(1 / scale.x, 1 / scale.y, 1 / scale.z)
+end
+
+function getButtonIndex(obj, params)
+    local buttons = obj.getButtons()
+    if buttons == nil then
+        return nil
+    end
+    for _,button in pairs(buttons) do
+        local match = true
+        for key,val in pairs(params) do
+            if button[key] ~= val then
+                match = false
+                break
+            end
+        end
+        if match then
+            return button.index
+        end
+    end
+    return nil
+end
+
+function setButton(obj, params)
+    local identifier = {click_function = params.click_function, function_owner = params.function_owner}
+    local buttonIndex = getButtonIndex(obj, identifier)
+    if not buttonIndex then
+        obj.createButton(params)
+    else
+        local newParams = {index = buttonIndex}
+        for k,v in pairs(params) do
+            newParams[k] = v
+        end
+        obj.editButton(newParams)
+    end
+end
+
+function removeButton(obj, params)
+    local identifier = {click_function = params.click_function, function_owner = params.function_owner}
+    local buttonIndex = getButtonIndex(obj, identifier)
+    if buttonIndex then
+        obj.removeButton(buttonIndex)
+    end
+end
+
+function setSelfButton(params)
+    local buttonParams = {
+        function_owner = self,
+        position       = Vector(0.6, 1, -1.1),
+        rotation       = Vector(0,0,0),
+        scale          = Vector(0.2,0.2,0.2),
+        width          = 2000,
+        height         = 500,
+        font_size      = 300,
+    }
+    for k,v in pairs(params) do
+        buttonParams[k] = v
+    end
+    if not self.getButtons() then
+        self.createButton(buttonParams)
+    else
+        buttonParams.index = 0
+        self.editButton(buttonParams)
+    end
+end
+
+function getZoneColor(zoneGUID)
+    for color,selected in pairs(Global.getVar("selectedColors")) do
+        if selected.zone.guid == zoneGUID then
+            return color
+        end
+    end
+end
+
+function getZoneGUID(color)
+    if color == nil then
+        return nil
+    end
+    local selected = Global.getVar("selectedColors")[color]
+    if selected == nil then
+        return nil
+    end
+    return selected.zone.guid
+end
+
+function getTablePos(zoneGUID)
+    return Global.getVar("playerTables")[getZoneColor(zoneGUID)].getPosition()
+end
+
+function getSpiritName(zoneGUID)
+    for _,obj in pairs(getObjectFromGUID(zoneGUID).getObjects()) do
+        if obj.hasTag("Spirit") then
+            return obj.getName()
+        end
+    end
```

Comment:
does this function not need a default return at the end? or is there an implied nil?

Reply:
I've added a default return.